**1. Среднее
Дано:** 3 случайных числа.

**Задание:** написать программу, которая будет вычислять среднее значение этих чисел.

**Пример:** (52 + 56 + 60) / 3 = 56

In [1]:
#1. The mean
from random import randint
def mean(nums):
    print(nums)
    print(f"The mean is = {sum(nums) / len(nums)}\n")
nums = []
for index in range(3):
    nums.append(randint(0, 100))
mean(nums)

#2. Division
import random
x, y = [random.random()*(1000**random.random()) for i in range(2)]
print(f"Random numbers:\n{x=}\n{y=}")
print(f"Integer= {x//y}")
print(f"Remainder= {x%y}\n")


#3 Rounding
a = 14.721
print(f"{a:.2f}")
print(f"{round(a)}")
print(f"{a:011}")

[73, 14, 80]
The mean is = 55.666666666666664

Random numbers:
x=56.77073374597798
y=111.7553892980183
Integer= 0.0
Remainder= 56.77073374597798

14.72
15
0000014.721


**2. Деление и еще раз деление**

**Дано:** 2 случайных числа.

**Задание:** написать программу, которая будет печать результат целочисленного деления этих чисел, а также остаток от деления первого от второго.

**Пример:** 

x = 177 и y = 10

**Результат:**
 17, 7

In [2]:
#2. Division
import random
x, y = [random.random()*(1000**random.random()) for i in range(2)]
print(f"Random numbers:\n{x=}\n{y=}")
print(f"Integer= {x//y}")
print(f"Remainder= {x%y}\n")

Random numbers:
x=109.97346873721011
y=22.29419679760813
Integer= 4.0
Remainder= 20.79668154677759



**3. Округление**
**Дано:** число с плавающей точкой.

**Задание:** написать программу, которая будет округлять заданное число:

1) 2х знаков после запятой; 2) до целого; 3) дополнять слева столько нулей, сколько не хватает числу до 11 знаков.

**Пример:**

x = 14.721

1) 14.72

2) 15

3) 0000014.721

In [3]:
#3 Rounding
a = 14.721
print(f"{a:.2f}")
print(f"{round(a)}")
print(f"{a:011}")

14.72
15
0000014.721


**[Junior] 4. Число "наоборот"**

**Дано:** целое число (int).

**Задание:** написать программу, которая будет инвертировать целое число

**Пример:**

 1)x = 123 -> 321 2) x = -325 -> -523 3) x = 0 -> 0 4) x = 1563847412 -> 0

In [3]:
#4-5. Invert integer
def reverse_number(number):
    rev_number = 0
    while number > 0:
        x = number % 10
        number = number // 10
        rev_number = rev_number * 10
        rev_number = rev_number + x
    if (rev_number.bit_length() / 8) < 4:
        print(f"-> {rev_number}")
    else:
        print("-> 0")
number1 = 123
reverse_number(number1)
number2 = 1563847412
reverse_number(number2)

-> 321
-> 0
